product_configurator_wizard/wizard/product_configurator.py

```diff
@@ -361,17 +430,40 @@ def fields_view_get(self, view_id=None, view_type='form',
             k: v for k, v in fields.iteritems() if k.startswith(
                 self.field_prefix) or k.startswith(self.custom_field_prefix)
         }
+        support_fields = {
+            k: v for k, v in fields.iteritems() if
+            k.startswith(self.ro_field_prefix) or
+            k.startswith(self.reqd_field_prefix) or
+            k.startswith(self.invis_field_prefix)
+        }
 
         res['fields'].update(dynamic_fields)
-        mod_view = self.add_dynamic_fields(res, dynamic_fields, wiz)
+        res['fields'].update(support_fields)
+        mod_view = self.add_dynamic_fields(
+            res, dynamic_fields, support_fields, wiz)
 
         # Update result dict from super with modified view
         res.update({'arch': etree.tostring(mod_view)})
 
+        wiz_vals = wiz.with_context(from_fvg=True).read(
+            dynamic_fields.keys())[0]
+        dynamic_field_vals = {
+            k: wiz_vals.get(
+                k, [] if v['type'] == 'many2many' else False
+                )
+            for k, v in fields.iteritems()
+            if k.startswith(self.field_prefix)
+        }
+        domains = {k: dynamic_fields[k]['domain']
+                   for k in dynamic_field_vals.keys()}
+        cfg_step = wiz.get_cfg_step()
+        vals = wiz.get_form_vals(dynamic_field_vals, domains, cfg_step)
+        if vals:
+            wiz.write(vals)
```

Comment:
This structure actually came about from setting defaults, and when this is merged and I rebase that one, it may make more sense.  The vals returned may have defaults.